testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/logging/AbstractLoggingTestCase.java

```diff
@@ -0,0 +1,330 @@
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.wildfly.core.testrunner.ServerController;
+
+/**
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+public abstract class AbstractLoggingTestCase {
+
+    public static final String DEPLOYMENT_NAME = "logging-test.jar";
+
+    public static final PathElement SUBSYSTEM_PATH = PathElement.pathElement(ModelDescriptionConstants.SUBSYSTEM, "logging");
+
+    public static final PathAddress SUBSYSTEM_ADDRESS = PathAddress.pathAddress(SUBSYSTEM_PATH);
+
+    @Inject
+    protected ServerController container;
+
+    protected static ModelControllerClient client;
+
+    @BeforeClass
+    public static void configureClient() {
+        client = TestSuiteEnvironment.getModelControllerClient();
+    }
+
+    static JavaArchive createDeployment() {
+        return createDeployment(LoggingServiceActivator.class, LoggingServiceActivator.DEPENDENCIES);
+    }
+
+    static JavaArchive createDeployment(final Map<String, String> manifestEntries) {
+        return createDeployment(LoggingServiceActivator.class, manifestEntries, LoggingServiceActivator.DEPENDENCIES);
+    }
+
+    static JavaArchive createDeployment(final Class<? extends ServiceActivator> serviceActivator, final Class<?>... classes) {
+        return createDeployment(serviceActivator, Collections.<String, String>emptyMap(), classes);
+    }
+
+    static JavaArchive createDeployment(final Class<? extends ServiceActivator> serviceActivator,
+                                        final Map<String, String> manifestEntries, final Class<?>... classes) {
+        final JavaArchive archive = ShrinkWrap.create(JavaArchive.class, DEPLOYMENT_NAME);
+        archive.addClasses(classes);
+        archive.addAsServiceProviderAndClasses(ServiceActivator.class, serviceActivator);
+        boolean addDeps = true;
+        final StringBuilder manifest = new StringBuilder();
+        for (String key : manifestEntries.keySet()) {
+            if ("Dependencies".equals(key)) {
+                addDeps = false;
+                manifest.append(key)
+                        .append(": ")
+                        .append("io.undertow.core,")
+                        .append(manifestEntries.get(key))
+                        .append('\n');
+            } else {
+                manifest.append(key)
+                        .append(": ")
+                        .append(manifestEntries.get(key))
+                        .append('\n');
+            }
+        }
+        if (addDeps) {
+            manifest.append("Dependencies: io.undertow.core");
+        }
+        archive.addAsResource(new StringAsset(manifest.toString()), "META-INF/MANIFEST.MF");
+        return archive;
+    }
+
+    public static ModelNode executeOperation(final ModelNode op) throws IOException {
```

Comment:
No real reason other than I just copied this over from the subsystem tests some time ago.